pandas/core/algorithms.py

```diff
@@ -204,6 +204,10 @@ def value_counts(values, sort=True, ascending=False, normalize=False,
     is_period = com.is_period_arraylike(values)
     values = Series(values).values
     is_category = com.is_categorical_dtype(values.dtype)
+    if is_category and dropna:
+        values = values.dropna()
+        if np.nan in values.categories:
```

Comment:
Should `Categorical.dropna` use `np.nan in self.categories`, or is there a better way to do it?